CMA 2.12.1-376 Release Notes (#73952)
openshift/openshift-docs: nodes/cma/nodes-cma-autoscaling-custom-rn.adoc

|{product-title} version
|General availability

|2.12.1
|4.15
|General availability

|2.12.1
|4.14
|General availability

|2.12.1
|4.13
|General availability

|2.12.1
|4.12
|General availability
|===

[id="nodes-pods-autoscaling-custom-rn-2121-376_{context}"]
== Custom Metrics Autoscaler Operator 2.12.1-376 release notes

This release of the Custom Metrics Autoscaler Operator 2.12.1-376 provides security updates and bug fixes for running the Operator in an {product-title} cluster. The following advisory is available for the link:https://access.redhat.com/errata/RHSA-TBD[RHSA-TBD].

[IMPORTANT]
====
Before installing this version of the Custom Metrics Autoscaler Operator, remove any previously installed Technology Preview versions or the community-supported version of KEDA.
====

[id="nodes-pods-autoscaling-custom-rn-2121-376-bugs_{context}"]
=== Bug fixes

> * Previously, if invalid values, such as nonexistent namespaces, were specified in scaled object metadata, the underling scaler clients would not free/close their client descriptors, resulting in a slow memory leak. This fix properly closes the underlying client descriptors when there are errors, preventing memory from leaking. (link:https://issues.redhat.com/browse/OCPBUGS-30145[*OCPBUGS-30145*])

Reviewer: hahaha apparently I typed "underling" the first time, sorry. As funny as that would have been... :smile: 
```suggestion
* Previously, if invalid values, such as nonexistent namespaces, were specified in scaled object metadata, the underlying scaler clients would not free/close their client descriptors, resulting in a slow memory leak. This fix properly closes the underlying client descriptors when there are errors, preventing memory from leaking. (link:https://issues.redhat.com/browse/OCPBUGS-30145[*OCPBUGS-30145*])
```

Author: @jkyros One of those typos that passes the spell checker! Thanks.

[id="nodes-pods-autoscaling-custom-rn-2112-332-bugs_{context}"]
=== Bug fixes

> * Because the Custom Metrics Autoscaler Operator version 3.11.2-311 was released without a required volume mount in the Operator deployment, the Custom Metrics Autoscaler Operator pod would restart every 15 minutes. This fix adds the required volume mount to the operator deployment. As a result, the Operator no longer restarts every 15 minutes. (link:https://issues.redhat.com/browse/OCPBUGS-22361[*OCPBUGS-22361*])

Reviewer 2: ```suggestion
* Because the Custom Metrics Autoscaler Operator version 3.11.2-311 was released without a required volume mount in the Operator deployment, the Custom Metrics Autoscaler Operator pod would restart every 15 minutes. This fix adds the required volume mount to the Operator deployment. As a result, the Operator no longer restarts every 15 minutes. (link:https://issues.redhat.com/browse/OCPBUGS-22361[*OCPBUGS-22361*])
```

> * Previously the `ServiceMonitor` custom resource (CR) for the `keda-metrics-apiserver` pod was not functioning, as it was referencing an incorrect metrics port name of "http". This fix corrects the `ServiceMonitor` CR to reference the proper port name of "metrics". As a result, the Service Monitor functions properly. (link:https://issues.redhat.com/browse/OCPBUGS-25806[*OCPBUGS-25806*])

Reviewer 3: ```suggestion
* Previously the `ServiceMonitor` custom resource (CR) for the `keda-metrics-apiserver` pod was not functioning, because it referenced an incorrect metrics port name of "http". This fix corrects the `ServiceMonitor` CR to reference the proper port name of "metrics". As a result, the Service Monitor functions properly. (link:https://issues.redhat.com/browse/OCPBUGS-25806[*OCPBUGS-25806*])
```

> * Previously the `ServiceMonitor` custom resource (CR) for the `keda-metrics-apiserver` pod was not functioning, because the CR referenced an incorrect metrics port name of "http". This fix corrects the `ServiceMonitor` CR to reference the proper port name of "metrics". As a result, the Service Monitor functions properly. (link:https://issues.redhat.com/browse/OCPBUGS-25806[*OCPBUGS-25806*])

Reviewer 2: ```suggestion
* Previously the `ServiceMonitor` custom resource (CR) for the `keda-metrics-apiserver` pod was not functioning, because the CR referenced an incorrect metrics port name of `http`. This fix corrects the `ServiceMonitor` CR to reference the proper port name of `metrics`. As a result, the Service Monitor functions properly. (link:https://issues.redhat.com/browse/OCPBUGS-25806[*OCPBUGS-25806*])
```
